Solution for issue #2434 : Removed scanning messages if you scan your own ships

EndrosG opened the pull request:
Removed scanning messages if you scanned one of your own ships, solving issue #2434

endless-sky commented on source/Ship.cpp:
```diff
@@ -1439,18 +1439,18 @@ int Ship::Scan()
 		Audio::Play(Audio::Get("scan"), Position());
 	
-	if(startedScanning && government->IsPlayer())
+	if(startedScanning && government->IsPlayer() && !target->GetGovernment()->IsPlayer())
```
I'd argue that if you try to scan one of your own ships (for whatever reason), you should still get a confirmation of that via a message.

Escorts will never scan anything, so if a player ship is starting a scan, that ship is guaranteed to be the flagship.

MessyMix replied:
Sure. But referring to your own flagship as an "Escort ship" is just... well... incorrect. :P

endless-sky replied:
If you look at the line of code following this one, it actually says, "Attempting to scan the ship <name>." It's not one of the messages that mentions the government name.

MessyMix replied:
Oh, you were referring to initiating a scan. Sorry, I glossed over that aspect. The outcome of the scan should not include the Escort government, though.

EndrosG commented:
Okay, how do you want it then?
My thought was to remove the messages completely - including the attempt notification as well, not only the success notification which raised the issue #2434.
We would also easily put in "better" messages on success, i.e. two different messages for each scan success.
I am open to any of the ways.

KiLEdEnNis commented:
modifying the message to "You are scanning ~" could work

endless-sky commented:
I was just commenting on that particular line. The changes in the other three lines following that make sense, but I think that initiating a scan of one of your escorts should still display the "Attempting to scan..." message, because it would be confusing to players if it did not (especially if they have sound off, and thus have no other indication that a scan is in progress).

EndrosG commented:
Okay, I changed the first thing back.

I removed the ` && !target->GetGovernment()->IsPlayer()` in line 1442 , and thus could also remove the ` && !government->IsPlayer()` in line 1444 since that was only needed after adding the first one.
